### PR TITLE
Add "Add type annotation to parameter" refactoring + remove dangling file lost in recent refactorings

### PR DESCRIPTION
**Pull request by cartermp**
Quick sample of it so far:

image

image

- [x] There are several TODOs to work out here before this is reviewable
- [x] Tests

### CODE REVIEW

**Comment by cartermp** [comment]
A note:

This only applies to parameters for now. Why? Because of `GetSymbolUseAtLocation`.

`GetSymbolUseAtLocation` constructs an `FSharpSymbolUse` on the fly rather than grabbing existing ones in our name resolution sink. Because of this, it also encodes an assumption that the `ItemOccurence` of the `FSharpSymbolUse` is `Use`. This is actually incorrect when the `FSharpSymbolUse` in question is a let-bound value or parameter. The `ItemOccurence` for these should be `Binding`. But it's not.

There are three alternatives I see here:

1. Use `GetAllUsesOfAllSymbolsInFile` and filter by range. This gives us the _correct_ `FSharpSymbolUse` and then lets us apply this to all kinds of `Binding`.
2. Call `GetSymbolAtLocation` and then pass it to a call of `GetUsesOfSymbolInFile` and filter by range.
3. Change `GetSymbolUseAtLocation` to pull data from the symbol use cache we have. However, this cache is just a `TcSymbolUseData [][]`, so this would be just as inefficient as if we just called `GetAllUsesOfAllSymbolsInFile`. So doing this would imply that we change our cache of these objects to allow indexing by range or some other thing.

I don't really want to entertain any of them in this PR. I think this PR accomplishes the majority use case anyways.

File: vsintegration/src/FSharp.Editor/Refactor/AddExplicitTypeToParameter.fs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                        if not (Char.IsWhiteSpace(ch)) then
+                            ch = '('
+                        else
+                            leftLoop sourceText.[pos - 1] (pos - 1)
+
+                    let rec rightLoop ch pos =
+                        if not (Char.IsWhiteSpace(ch)) then
+                            ch = ')'
+                        else
+                            leftLoop sourceText.[pos + 1] (pos + 1)
```
Comment (cartermp) [inline vsintegration/src/FSharp.Editor/Refactor/AddExplicitTypeToParameter.fs:73]:
```suggestion
                            rightLoop sourceText.[pos + 1] (pos + 1)
```

**Comment by cartermp** [comment]
ugh I need to tweak the surface changes

File: vsintegration/src/FSharp.Editor/Refactor/AddExplicitTypeToParameter.fs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
```
Comment (cartermp) [inline vsintegration/src/FSharp.Editor/Refactor/AddExplicitTypeToParameter.fs:76]:
```suggestion
                            rightLoop sourceText.[pos + 1] (pos + 1)
```

File: src/fsharp/symbols/Symbols.fsi [hunk 3]
```diff
@@ -900,6 +900,9 @@ type FSharpMemberOrFunctionOrValue =
 
     /// Indicated if this is a value compiled to a method
     member IsValCompiledAsMethod: bool
+
+    /// Indicates if this is a value
```
Comment (baronfel) [inline src/fsharp/symbols/Symbols.fsi:904]:
```suggestion
    /// Indicates if this is a function
```

**Review by TIHan — APPROVED** [review]
Looks good to me. The tests are great.

**Pull request merged**